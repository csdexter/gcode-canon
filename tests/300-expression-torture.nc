(testing expression parsing and evaluation)
(minimal context)
G21 G01 X3 F60

(following code taken from the official RS274NGC spec)
X [1 + 2] (X SHOULD BE 3)
X [1 - 2] (X SHOULD BE -1)
X [1 --3] (X SHOULD BE 4)
X [2/5] (X SHOULD BE 0.40)
X [3.0 * 5] (X SHOULD BE 15)
X [0 OR 0] (X SHOULD BE 0)
X [0 OR 1] (X SHOULD BE 1)
X [2 OR 2] (X SHOULD BE 1)
X [0 AND 0] (X SHOULD BE 0)
X [0 AND 1] (X SHOULD BE 0)
X [2 AND 2] (X SHOULD BE 1)
X [0 XOR 0] (X SHOULD BE 0)
X [0 XOR 1] (X SHOULD BE 1)
X [2 XOR 2] (X SHOULD BE 0)
X [15 MOD 4.0] (X SHOULD BE 3)
X [1 + 2 * 3 - 4 / 5] (X SHOULD BE 6.2)
X SIN[30] (X SHOULD BE 0.5)
X COS[0.0] (X SHOULD BE 1.0)
X TAN[60.0] (X SHOULD BE 1.7321)
X SQRT[3] (X SHOULD BE 1.7321)
X ATAN[1.7321]/[1.0] (X SHOULD BE 60.0)
X ASIN[1.0] (X SHOULD BE 90.0)
X ACOS[0.707107] (X SHOULD BE 45.0000)
X ABS[20.0] (X SHOULD BE 20)
X ABS[-1.23] (X SHOULD BE 1.23)
X ROUND[-0.499] (X SHOULD BE 0)
X ROUND[-0.5001] (X SHOULD BE -1.0)
X ROUND[2.444] (X SHOULD BE 2)
X ROUND[9.975] (X SHOULD BE 10)
X FIX[-0.499] (X SHOULD BE -1.0)
X FIX[-0.5001] (X SHOULD BE -1.0)
X FIX[2.444] (X SHOULD BE 2)
X FIX[9.975] (X SHOULD BE 9)
X FUP[-0.499] (X SHOULD BE 0.0)
X FUP[-0.5001] (X SHOULD BE 0.0)
X FUP[2.444] (X SHOULD BE 3)
X FUP[9.975] (X SHOULD BE 10)
X EXP[2.3026] (X SHOULD BE 10)
X LN[10.0] (X SHOULD BE 2.3026)
X [2 ** 3.0] #1=2.0 (X SHOULD BE 8.0)
##1 = 0.375 (#1 IS 2, SO PARAMETER 2 IS SET TO 0.375)
X #2 (X SHOULD BE 0.375) #3=7.0
#3=5.0 X #3 (PARAMETERS SET IN PARALLEL, SO X SHOULD BE 7, NOT 5)
X #3 #3=1.1 (PARAMETERS SET IN PARALLEL, SO X SHOULD BE 5, NOT 1.1)
X [2 + ASIN[1/2.1+-0.345] / [ATAN[FIX[4.4] * 2.1 * SQRT[16.8]] /[-18]]**2]
X SQRT[3**2 + 4**2] (X SHOULD BE 5.0)

(that's all, folks!)
M02
