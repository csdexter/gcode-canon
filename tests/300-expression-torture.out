MSG: WAR: Machine servos activated!
MSG: STA: Scanning input for programs (O words)
MPOS,3.00,0.00,0.00
MPOS,-1.00,0.00,0.00
MPOS,4.00,0.00,0.00
MPOS,0.40,0.00,0.00
MPOS,15.00,0.00,0.00
MPOS,0.00,0.00,0.00
MPOS,1.00,0.00,0.00
MPOS,0.00,0.00,0.00
MPOS,1.00,0.00,0.00
MPOS,0.00,0.00,0.00
MPOS,1.00,0.00,0.00
MPOS,0.00,0.00,0.00
MPOS,3.00,0.00,0.00
MPOS,6.20,0.00,0.00
MPOS,0.50,0.00,0.00
MPOS,1.00,0.00,0.00
MPOS,1.73,0.00,0.00
MPOS,60.00,0.00,0.00
MPOS,90.00,0.00,0.00
MPOS,45.00,0.00,0.00
MPOS,20.00,0.00,0.00
MPOS,1.23,0.00,0.00
MPOS,0.00,0.00,0.00
MPOS,-1.00,0.00,0.00
MPOS,2.00,0.00,0.00
MPOS,10.00,0.00,0.00
MPOS,-1.00,0.00,0.00
MPOS,2.00,0.00,0.00
MPOS,9.00,0.00,0.00
MPOS,0.00,0.00,0.00
MPOS,3.00,0.00,0.00
MPOS,10.00,0.00,0.00
MPOS,10.00,0.00,0.00
MPOS,2.30,0.00,0.00
MPOS,8.00,0.00,0.00
MPOS,0.38,0.00,0.00
MPOS,7.00,0.00,0.00
MPOS,5.00,0.00,0.00
MPOS,2.00,0.00,0.00
MPOS,5.00,0.00,0.00
